### PR TITLE
[Elixir] Improve concept exercise: basics

### PR DESCRIPTION
**Pull request by neenjaw**
Some improvements to the basics exercise' introduction and after document.

I have added some content re: `@doc` and `@moduledoc` as comment types (rather than as heredoc strings attached to module attributes).

### CODE REVIEW

File: languages/elixir/exercises/concept/basics/.docs/introduction.md [hunk 1]
```diff
@@ -1,44 +1,42 @@
...
 
+## Modules
+
 Elixir is an [functional-programming language][functional-programming] and requires all named functions to be defined in a _module_. The `defmodule` keyword is used to define a module. All modules are available to all other modules at runtime and do not require an _access modifier_ to make them visible to other parts of the program. A _module_ is analogous to a _class_ in other programming languages.
 
-```elixir
-defmodule Calculator do
-  # ...
-end
-```
```
Comment (angelikatyborska) [inline languages/elixir/exercises/concept/basics/.docs/introduction.md:23]: I would keep this example here. It's new syntax, it makes it easier to visualize what the paragraph is talking about it, and it makes the later code example that uses `defmodule` a bit easier to process.

File: languages/elixir/exercises/concept/basics/.docs/after.md [hunk 2]
```diff
@@ -1,51 +1,75 @@
...
-# add_3/1 because this function has one parameter, thus an arity of 1
-def add_3(x) do
-  3 + x
-end
-```
+- Elixir supports [inline documentation][inline-documentation]
 
-Elixir supports one type of comment for [inline documentation][inline-documentation]. Single line comments are preceded by `#`.
+  - Single line comments are preceded by `#`.
+  - Comments to document a named function must precede the named function definition
```
Comment (angelikatyborska) [inline languages/elixir/exercises/concept/basics/.docs/after.md:55]: I am not sure if I would put writing documentation with `@doc` and `@moduledoc` under the topic of "comments". In the docs, it's called "annotating" https://elixir-lang.org/getting-started/module-attributes.html#as-annotations

Maybe it makes sense to turn it around and write about "documentation", including inline comments as one of the methods? Something like in https://elixirschool.com/en/lessons/basics/documentation/

Comment (neenjaw) [reply languages/elixir/exercises/concept/basics/.docs/after.md:55]: That is true, I just didn't want to touch on module annotations just yet.  I will reorient it under the heading `Documentation` instead.

**Comment by neenjaw** [comment]
@angelikatyborska I have made some changes re: comments -> documentation.  Would be interested in your feedback before merging.

**Comment by angelikatyborska** [comment]
Great changes, exactly what I had in mind. Good to merge!

**Pull request merged**